src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -565,6 +567,32 @@ static bool has_non_debug_usages(Node* n) {
   return false;
 }
 
+static bool is_klass_initialized(Symbol* klass_name) {
+  VM_ENTRY_MARK;
+  InstanceKlass* ik = SystemDictionary::find_instance_klass(klass_name, Handle(), Handle());
+  return ik != nullptr && ik->is_initialized();
+}
+
+static bool is_box_cache_valid(CallNode* call) {
+  ciInstanceKlass* klass = call->as_CallStaticJava()->method()->holder();
+  BasicType box_type = klass->box_klass_type();
+
+  if (box_type != T_OBJECT) {
+    switch(box_type) {
+      case T_INT:     return is_klass_initialized(java_lang_Integer_IntegerCache::symbol());
+      case T_CHAR:    return is_klass_initialized(java_lang_Character_CharacterCache::symbol());
+      case T_SHORT:   return is_klass_initialized(java_lang_Short_ShortCache::symbol());
+      case T_BYTE:    return is_klass_initialized(java_lang_Byte_ByteCache::symbol());
+      case T_LONG:    return is_klass_initialized(java_lang_Long_LongCache::symbol());
+      case T_BOOLEAN:
+      case T_FLOAT:
+      case T_DOUBLE:  return true;
+      default:;
+    }
+  }
+  return false;
+}
+
```

Comment:
The code is correct but, please, move this new code into ciInstanceKlass (near other `box` queries). The VM queries should be done in CI.
```
static bool is_box_cache_valid(CallNode* call) {
  ciInstanceKlass* klass = call->as_CallStaticJava()->method()->holder();
  return klass->is_box_cache_valid();
}
```

Reply:
I have cleaned these codes. Thank you for your review.